src/common/ros2_util/yaml_convert.h

```diff
@@ -214,19 +214,34 @@ inline bool YamlToMessage(
   YAML::Node root;
   try {
     root = YAML::Load(yaml_str);
+  } catch (const YAML::ParserException& e) {
+    std::cerr << "YAML Parsing Error: " << e.what() << std::endl;
+    return false;  // Return false on YAML parsing error
+  } catch (const std::exception& e) {
+    std::cerr << "General Error: " << e.what() << std::endl;
+    return false;  // Catch general exceptions
   } catch (...) {
-    return false;
+    std::cerr << "Unknown error occurred during YAML parsing" << std::endl;
```

Comment:
Thank you for your contribution and the detailed error handling implementation! I really appreciate the effort to make the code more robust.

However, I have some concerns about using `std::cerr` for error reporting in this context:

1. Error messages written to `std::cerr` cannot be captured by our logging system, making it difficult to track and debug issues in production environments

2. `std::cerr` is not thread-safe, which could lead to garbled output in multi-threaded scenarios